cmake/utils.cmake

```diff
@@ -0,0 +1,369 @@
+# This file is taken from github.com/vllm-project/vllm/cmake/utils.cmake
+# It contains utility functions for building PyTorch extensions with GPU support.
+# The only modification is finding the `Python` package with a specific version.
+#
+# THIS MACRO WAS MODIFIED
+# Attempt to find the python package and verifies that it is one of the `SUPPORTED_VERSIONS`.
+# To customize which python gets found, set the `Python_EXECUTABLE` variable.
+macro (find_python_constrained_versions SUPPORTED_VERSIONS)
```

Comment:
I forgot to comment about this on #17 but the old process for finding the python version based off the executable was very intentional.  Some of the CI systems had multiple versions of python installed and it was important to select the specific version requested rather than just the first in the list of supported verions.